hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -305,6 +301,247 @@ async def attempts(self):
+                yield job
+            last_job_id = body.get('last_job_id')
+            if last_job_id is None:
+                break
+
+    async def status(self) -> GetJobGroupResponseV1Alpha:
+        self._raise_if_not_submitted()
+        resp = await self._client._get(f'/api/v1alpha/batches/{self.batch_id}/job-groups/{self.job_group_id}')
+        json_status = await resp.json()
+        assert isinstance(json_status, dict), json_status
+        self._last_known_status = cast(GetJobGroupResponseV1Alpha, json_status)
+        return self._last_known_status
+
+    async def last_known_status(self) -> GetJobGroupResponseV1Alpha:
+        self._raise_if_not_submitted()
+        if self._last_known_status is None:
+            return await self.status()  # updates _last_known_status
+        return self._last_known_status
+
+    def create_job(self, image: str, command: List[str], **kwargs) -> Job:
+        return self._batch._create_job(self, {'command': command, 'image': image, 'type': 'docker'}, **kwargs)
+
+    def create_jvm_job(self, jar_spec: Dict[str, str], argv: List[str], *, profile: bool = False, **kwargs):
+        return self._batch._create_job(
+            self, {'type': 'jvm', 'jar_spec': jar_spec, 'command': argv, 'profile': profile}, **kwargs
+        )
+
+    def create_job_group(
+        self,
+        *,
+        attributes: Optional[Dict[str, str]] = None,
+        callback: Optional[str] = None,
+        cancel_after_n_failures: Optional[int] = None,
+    ) -> 'JobGroup':
+        return self._batch._create_job_group(
+            self,
+            attributes=attributes,
+            callback=callback,
+            cancel_after_n_failures=cancel_after_n_failures,
+        )
+
+    # FIXME Error if this is called while in a job within the same job group
+    async def _wait(
+        self,
+        description: str,
+        progress: BatchProgressBar,
+        disable_progress_bar: bool,
+    ) -> GetJobGroupResponseV1Alpha:
+        self._raise_if_not_submitted()
+        deploy_config = get_deploy_config()
+        url = deploy_config.external_url('batch', f'/batches/{self.batch_id}')
+        i = 0
+        status = await self.status()
+
+        if is_notebook():
+            description += f'[link={url}]{self.batch_id}[/link]'
+        else:
+            description += url
+
+        with progress.with_task(description, total=status['n_jobs'], disable=disable_progress_bar) as progress_task:
+            while True:
+                status = await self.status()
+                progress_task.update(None, total=status['n_jobs'], completed=status['n_completed'])
+                if status['complete']:
+                    return status
+                j = random.randrange(math.floor(1.1**i))
+                await asyncio.sleep(0.100 * j)
+                # max 44.5s
+                if i < 64:
+                    i = i + 1
+
+    # FIXME Error if this is called while in a job within the same job group
```

Comment:
Resolve the FIXME or create an issue.

Reply:
https://github.com/hail-is/hail/issues/14338